src/ore/src/str.rs

```diff
@@ -83,6 +98,36 @@ impl<'a> Deref for QuotedStr<'a> {
     }
 }
 
+/// Same as [`QuotedStr`], but also escapes new lines and tabs.
+///
+/// Constructed by [`StrExt::escaped`].
+#[derive(Debug)]
+pub struct EscapedStr<'a>(&'a str);
+
+impl<'a> fmt::Display for EscapedStr<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.write_char('"')?;
+        for c in self.chars() {
+            match c {
+                '"' => f.write_str("\\\"")?,
+                '\n' => f.write_str("\\n")?,
+                '\r' => f.write_str("\\r")?,
+                '\t' => f.write_str("\\t")?,
+                _ => f.write_char(c)?,
+            }
+        }
+        f.write_char('"')
+    }
+}
```

Comment:
I'm wondering if it'd be better to use Rust's built-in functionality for escaping characters: https://doc.rust-lang.org/std/primitive.char.html#method.escape_default

Reply:
Ok, then the manual version is fine. We indeed want to render unicode as is.

But please add a code comment that it's not using the built-in functionality, because somebody reading this code at some later point might wonder about the same thing.

Reply:
Done, thanks!